backend/app/data_layer/database/crud/postgresql/user_crud.py

```
@@ -0,0 +1,277 @@
+        The user object to add to the database
+    session: ``Session ``
+        Session object to interact with the database
+    """
+    try:
+        session.add(user)
+        session.commit()
+        session.refresh(user)
+    except Exception as e:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST, f"Error while creating user: {e}"
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)